workflow/executor/executor_test.go

```diff
@@ -194,3 +195,62 @@ func TestIsTarball(t *testing.T) {
 		assert.Equal(t, test.isTarball, ok, test.path)
 	}
 }
+
+func TestChmod(t *testing.T) {
+	TmpDirName := "testdata/tmpdir"
```

Comment:
this is the right direction. I think you should use `ioutil.TempDir` to create the directory and create a subdirectory and file in that dir and then verify.

Reply:
Thanks for the hint 😄

Reply:
Great success!